3.2.0 seems to cause a build error in Angular/React: An export assignment cannot be used in a module with other exported elements.

### Background & Context

I tried to upgrade to latest v3.2.0 that drops the need for the `@types/dompurify` which is awesome, however my Angular build is now failing with this error (you can see the error in my CI workflow [failure](https://github.com/ghiscoding/Angular-Slickgrid/actions/runs/11803253896/job/32880801702))

> Error: node_modules/dompurify/dist/purify.cjs.d.ts:401:1 - error TS2309: An export assignment cannot be used in a module with other exported elements.
> 
> 401 export = _default;

If we browse to the CJS dist bundle from [unpkg](https://unpkg.com/browse/dompurify@3.2.0/dist/purify.cjs.d.ts), we can see that it's the last line of the file export that it doesn't seem to like. I don't fully understand the problem but it might be due to trying to export DOMPurify twice? Though again I'm not entirely sure, see this Stack Overflow [answer](https://stackoverflow.com/a/48708568/1212166) for that error

### Bug

Also another possible issue that I found on top of what is described above, is that I was a bit surprised that it picked up the CJS (CommonJS) build instead of the expect ESM dist. Taking a look at the `package.json`, it is most probably because the CJS `require` is the first export to appear, however it should really be the `import` that comes first and make the `require` a fallback (2nd) which is the inverse of what is currently set. Note that I could provide a PR for the correct order but I'm not really that will fix the 1st error above

https://github.com/cure53/DOMPurify/blob/2cf6e2510699261a0004e3431c4e8a0bbd65b230/package.json#L29-L38

We can see the correct order in NodeJS [conditional exports](https://nodejs.org/api/packages.html#conditional-exports)

> Within the ["exports"](https://nodejs.org/api/packages.html#exports) object, key order is significant. During condition matching, earlier entries have higher priority and take precedence over later entries. The general rule is that conditions should be from most specific to least specific in object order.

#### Input

n/a

#### Given output

n/a

#### Expected output

no build error

### Feature

n/a

Hi, I got the following errors with DOMPurify 3.2.0

```
>npx tsc

node_modules/dompurify/dist/purify.es.d.mts:170:28 - error TS2304: Cannot find name 'TrustedTypePolicy'.

170     TRUSTED_TYPES_POLICY?: TrustedTypePolicy | undefined;
                               ~~~~~~~~~~~~~~~~~

node_modules/dompurify/dist/purify.es.d.mts:245:9 - error TS2304: Cannot find name 'TrustedHTML'.

245     }): TrustedHTML;
            ~~~~~~~~~~~

node_modules/dompurify/dist/purify.es.d.mts:396:34 - error TS2339: Property 'trustedTypes' does not exist on type 'Window & typeof globalThis'.

396     trustedTypes?: typeof window.trustedTypes;
                                     ~~~~~~~~~~~~

src/mjs/dompurify.js:5:1 - error TS9006: Declaration emit for this file requires using private name 'DOMPurify' from module '"C:/Users/XXX/Documents/YYY/node_modules/dompurify/dist/purify.es"'. An explicit type annotation may unblock declaration emit.

5 import DOMPurify from 'dompurify';
  ~~~~~~

Found 4 errors in 2 files.
```

Additional note:
``` console
src/mjs/dompurify.js:5:1 - error TS9006: Declaration emit for this file requires using private name 'DOMPurify' from module '"C:/Users/XXX/Documents/YYY/node_modules/dompurify/dist/purify.es"'. An explicit type annotation may unblock declaration emit.

5 import DOMPurify from 'dompurify';
  ~~~~~~
```

I needed to fix import like below:
``` javascript
import { default as DOMPurify } from 'dompurify';
```

--- a/scripts/fix-cjs-types.js
+++ b/scripts/fix-cjs-types.js
@@ -22,7 +22,7 @@ const path = require('node:path');
   // Append `export = _default;` to match the
   // `module.exports = DOMPurify` statement
   // that Rollup creates.
-  fixed += '\nexport = _default;\n';
+  fixed += '\n// @ts-ignore\nexport = _default;\n';
 
   await fs.writeFile(fileName, fixed);
 })().catch((ex) => {

